src/main/java/com/couchbase/lite/store/SQLiteStore.java

```diff
@@ -145,7 +146,7 @@ public SQLiteStore(String directory, Manager manager, StoreDelegate delegate) {
         }
         this.path = new File(directory, kDBFilename).getPath();
         this.manager = manager;
-        this.storageEngine = null;
+        this.storageEngine = createStorageEngine();
```

Comment:
close() method might set null for storageEngine. So open() method needs to check if storageEngine is null.
I think storageEngine initialization is not necessary in Constructor.